src/midifunctor.cpp

```diff
@@ -828,7 +828,7 @@ FunctorCode GenerateMIDIFunctor::VisitNote(const Note *note)
             const double stopTime = startTime + midiNote.duration;
 
             m_midiFile->addNoteOn(m_midiTrack, startTime * tpq, channel, midiNote.pitch, velocity);
-            m_midiFile->addNoteOff(m_midiTrack, stopTime * tpq, channel, midiNote.pitch);
+            m_midiFile->addNoteOff(m_midiTrack, stopTime * tpq - 1, channel, midiNote.pitch);
```

Comment:
Can `stopTime * tpq` ever be 0? Is it a problem if you have `0 - 1`?

Reply:
Good point! I think it's highly unlikely since we're dealing with linearly increasing time, but in the absence of unit tests, better be safe than sorry. I added `std::max(0.0)` in front of each expression.